utils/src/main/scala/za/co/absa/enceladus/utils/transformations/DeepArrayTransformations.scala

```diff
@@ -44,14 +166,20 @@ object DeepArrayTransformations {
     * @param inputColumnName  A column name for which to apply the transformation, e.g. `company.employee.firstName`.
     * @param outputColumnName The output column name. The path is optional, e.g. you can use `conformedName` instead of `company.employee.conformedName`.
     * @param expression       A function that applies a transformation to a column as a Spark expression.
-    * @return A dataframe with a new field that contains transformed values.
+    * @param errorCondition   A function that should check error conditions and return an error column in case such conditions are met
+    * @param isAddColumn      If true the input column is not expected to exist, it should be added (this affects only exception thrown if the column already exists)
+    * @return A pair consisting of a dataframe with a new field that contains transformed values and a string containing the error column name.
     */
-  def nestedWithColumnMap(df: DataFrame,
-                          inputColumnName: String,
-                          outputColumnName: String,
-                          expression: Column => Column): DataFrame = {
+  private def nestedWithColumnMapHelper(df: DataFrame,
+                                        inputColumnName: String,
+                                        outputColumnName: String,
+                                        expression: Option[Column => Column] = None,
+                                        errorCondition: Option[Column => Column] = None,
+                                        isAddColumn: Boolean = false
```

Comment:
Suggestion/idea: I see this Boolean flag only affects the message of exception eventually thrown. And is used as `true` only on e place. Maybe omitting the flag, catching and re-raising the exception with the desired message on that one place would make the code less cumbersome.